src/validator.ts

```diff
@@ -680,6 +682,51 @@ function doIntrinsicGetAZs(ref: any, key: string){
 
 }
 
+function doIntrinsicSelect(ref: any, key: string){
+  let toGet = ref[key];
+  if(!Array.isArray(toGet) || toGet.length < 2) {
+    addError('crit', "Fn::Select only supports an array of two elements", placeInTemplate, "Fn::Select");
+    return 'INVALID_SELECT';
+  }
+  if (toGet[0] === undefined || toGet[0] === null) {
+     addError('crit', "Fn::Select first element cannot be null or undefined", placeInTemplate, "Fn::Select");
+     return 'INVALID_SELECT';
+  }
+
+  let index = parseInt(toGet[0]);
+  if (typeof index !== 'number' || isNaN(index)) {
+    addError('crit', "First element of Fn::Select must be a number", placeInTemplate, "Fn::Select");
+    return 'INVALID_SELECT';
+  }
+  if (toGet[1] === undefined || toGet[1] === null) {
+     addError('crit', "Fn::Select Second element cannot be null or undefined", placeInTemplate, "Fn::Select");
+     return 'INVALID_SELECT';
+  }
+
+  let list = toGet[1]
+  if (!Array.isArray(list)) {
+    //we may need to resolve it
+    if (typeof list !== 'string') {
+      list = resolveIntrinsicFunction(list, Object.keys(list)[0]);
```

Comment:
There should be some validation around resolving an intrinsic function, this is to support nested functions, see https://github.com/martysweet/cfn-lint/blob/master/src/validator.ts#L894-L897 for an example. This file will need updating with the list of the supported functions from AWS. https://github.com/martysweet/cfn-lint/blob/master/data/aws_intrinsic_functions.json#L13

```
Supported Functions
For the Fn::Select index value, you can use the Ref and Fn::FindInMap functions.

For the Fn::Select list of objects, you can use the following functions:

Fn::FindInMap

Fn::GetAtt

Fn::GetAZs

Fn::If

Fn::Split

Ref
```

Reply:
Thanks for explaining that bit.